### PR TITLE
 board/common : Add readme for flash partition

### PR DESCRIPTION
**Pull request by YiHyunJin**
Describes the FLASH PARTITION when porting a new board.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [7575afbce352881ddba378b3510416973c5514ab] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7575afbce352881ddba378b3510416973c5514ab] - Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@YiHyunJin As we discussed, let's reorganize this.

**Comment by seinfra** [comment]
Target : [9b2c4c54faf25b7d3e5ac476059bccaef88cfdb5] -

**Comment by seinfra** [comment]
Target : [c28929fb61a7960c2c4180f9c650bb10c8be581c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c28929fb61a7960c2c4180f9c650bb10c8be581c] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3ea345d554046946d8f7e3323eceefbf87d4cd83] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3ea345d554046946d8f7e3323eceefbf87d4cd83] - Code Rule Check (C++) OK.

File: os/board/common/README.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+os      | none    | 1536
+factory | none    | 1536
+user    | smartfs | 1400
+nvram   | config  | 8
+sssrw   | none    | 512
+
+
+## Partition size
+
+FLASH_PART_SIZE is Flash partition size. The unit of partition size is kb.
```
Comment (yashwanth686007) [inline os/board/common/README.md]: Better to mention the size as KB?

File: os/board/common/README.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        Hardware Configuration -> Board Selection -> append string at Flash partition type list
+        ```
+
+    3. Append "rom" at next of *user* to FLASH_PART_NAME
+
+        ```bash
+        Hardware Configuration -> Board Selection -> append string at FLash partition name list
+        ```
+
+    4. Enable the audomount config for romfs
```
Comment (yashwanth686007) [inline os/board/common/README.md]: "audomount" -> automount

File: os/board/common/README.md (hunk 1 above)
Comment (YiHyunJin) [reply os/board/common/README.md]: okay.. thanks

**Comment by seinfra** [comment]
Target : [5150a55c1918b264bbbc0da8e058cf6bfad897b5] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5150a55c1918b264bbbc0da8e058cf6bfad897b5] - Code Rule Check OK.

File: os/board/common/README.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
+# How to partition a flash into a config
```
Comment (jeongarmy) [inline os/board/common/README.md]: I think 'in' is right.

Comment (YiHyunJin) [reply os/board/common/README.md]: @jeongarmy Thanks. i will fix.

File: os/board/common/README.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+## Prepare
+
+To partition Flash, you must first enable ARCH_USE_FLASH. So if new board use FLASH, add 'select ARCH_USE_FLASH' to Kconfig. 
+
+## Contents
+
+> [How to Configure CONFIG](#how-to-configure-config)  
+> [Partition size](#Partition-size)  
+> [Partition type](#partition-type)  
+> [Partition name](#partition-name)
```
Comment (sunghan-chang) [inline os/board/common/README.md]: When do you use capital character?
Why are `Configure` and `size` different?

File: os/board/common/README.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+To partition Flash, you must first enable ARCH_USE_FLASH. So if new board use FLASH, add 'select ARCH_USE_FLASH' to Kconfig. 
+
+## Contents
+
+> [How to Configure CONFIG](#how-to-configure-config)  
+> [Partition size](#Partition-size)  
+> [Partition type](#partition-type)  
+> [Partition name](#partition-name)  
+
+## How to Configure CONFIG
```
Comment (sunghan-chang) [inline os/board/common/README.md]:
```
## How to use

Flash partition is provided by configurations, CONFIG_A, CONFIG_B and CONFIG_C.
CONFIG_A represents size of each partition, B does name, C does interface between flash and upper layer.
All partition information is separated by .....

### CONFIG_A
.....
### CONFIG_B
...
```

File: os/board/common/README.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+factory | none    | 1536
+user    | smartfs | 1400
+nvram   | config  | 8
+sssrw   | none    | 512
+
+
+## Partition size
+
+FLASH_PART_SIZE is Flash partition size. The unit of partition size is KB.  
+If you want to change partition size, you can change it at below location.
```
Comment (sunghan-chang) [inline os/board/common/README.md]: You can find and modify it using menuconfig as shown below:
'''
Board ...
'''

File: os/board/common/README.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+'config' provides an interface for saving and retrieving application configuration data to a standard MTD partition.  
+If you want to use config, you have to enable on the config device as shown below.  
+
+    File Systems -> Memory Technology Device (MTD) Support -> Enable Dev Config (MTD based) device -> Enable Dev Config (MTD based) device
+
+### smartfs
+
+> [How to Enable and use SmartFS](https://github.com/Samsung/TizenRT/blob/master/docs/HowToUseSmartFS.md#how-to-enable-and-use-smartfs)  
+
+## How to use romfs
```
Comment (sunghan-chang) [inline os/board/common/README.md]:
```
## APPENDIX
### How to set romfs partition
Here is an example to show how to set romfs partition on ARTIK05x.
```
1. Modify ...
```

File: os/board/common/README.md (hunk 4 above)
Comment (YiHyunJin) [reply os/board/common/README.md]: 'configure' is right. I will modify.

File: os/board/common/README.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
+# How to partition a flash into a config
+
+This document provides a way to register a partial block driver in Flash.
```
[the remark this answers is not in the document]
Comment (YiHyunJin) [reply os/board/common/README.md]: I will modify to 'This document provides a way to partition in Flash.'.

File: os/board/common/README.md (hunk 3 above)
Comment (YiHyunJin) [reply os/board/common/README.md]: okay. I will modify.

File: os/board/common/README.md (hunk 5 above)
Comment (YiHyunJin) [reply os/board/common/README.md]: ###CONFIG_A and ###CONFIG_B seem to be a overlap with ##partition size and ##Partition type.

File: os/board/common/README.md (hunk 6 above)
Comment (YiHyunJin) [reply os/board/common/README.md]: Okay. I will fix.

**Comment by seinfra** [comment]
Target : [b2cb1165e9a930163a5b1dd2a0e428789ccfb2b3] -

**Comment by seinfra** [comment]
Target : [b2cb1165e9a930163a5b1dd2a0e428789ccfb2b3] -

**Comment by seinfra** [comment]
Target : [d0d77d2b07c04a1b4458f260103b11f32df319d2] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [d0d77d2b07c04a1b4458f260103b11f32df319d2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [14df0d629b37b84697dc8dd0387de627543696bb] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [14df0d629b37b84697dc8dd0387de627543696bb] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9977bd3163cceba91fcaa1b6134d9eb7e7631ef1] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9977bd3163cceba91fcaa1b6134d9eb7e7631ef1] - Code Rule Check (C++) OK.

File: os/board/common/README.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
+# How to partition a flash
+
+This document provides a way to partition in flash. Use CONFIG when flash partition.
```
Comment (dr-venkman) [inline os/board/common/README.md]: Please rephrase as "This document describes the method to partition flash memory using TizenRT's menuconfig."

File: os/board/common/README.md [hunk 10]
```diff
@@ -0,0 +1,78 @@
+# How to partition a flash
+
+This document provides a way to partition in flash. Use CONFIG when flash partition.  
+
+## How to configure CONFIG
+
+Flash partition is provided by configurations, FLASH_PART_SIZE, FLASH_PART_TYPE and FLASH_PART_NAME. 
+FLASH_PART_SIZE represents size of each partition, FLASH_PART_SIZE does name, FLASH_PART_TYPE does interface between flash and upper layer.
```
Comment (dr-venkman) [inline os/board/common/README.md]: Please enumerate or Itemize the three configurations above as shown below:

"Flash partition can be configured the following three parameters:
1. FLASH_PART_SIZE -- represents a partition size
2. FLASH_PART_NAME (please check above) -- represents partition name
3. FLASH_PART_TYPE -- represents interface between flash and upper layer
"

File: os/board/common/README.md [hunk 11]
```diff
@@ -0,0 +1,78 @@
+# How to partition a flash
+
+This document provides a way to partition in flash. Use CONFIG when flash partition.  
+
+## How to configure CONFIG
+
+Flash partition is provided by configurations, FLASH_PART_SIZE, FLASH_PART_TYPE and FLASH_PART_NAME. 
+FLASH_PART_SIZE represents size of each partition, FLASH_PART_SIZE does name, FLASH_PART_TYPE does interface between flash and upper layer.  
+All partition information is separated by comma and the last must end with a comma. The information of FLASH_PART_SIZE, FLASH_PART_TYPE and FLASH_PART_NAME should be in order.
```
Comment (dr-venkman) [inline os/board/common/README.md]: The information of FLASH_PART_SIZE, FLASH_PART_TYPE and FLASH_PART_NAME should be in order.  --> "Note that the values for FLASH_PART_SIZE, FLASH_PART_TYPE and FLASH_PART_NAME must be specified in that order, as shown in the example below:"

File: os/board/common/README.md [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+sssro   | none    | 48
+bl2     | none    | 192
+sssfw   | none    | 512
+wlanfw  | none    | 2400
+os      | none    | 1536
+factory | none    | 1536
+user    | smartfs | 1400
+nvram   | config  | 8
+sssrw   | none    | 512
+
```
Comment (dr-venkman) [inline os/board/common/README.md]: Add a sentence here "In the following, we describe how to set the parameters using menuconfig."

File: os/board/common/README.md (hunk 9 above)
Comment (YiHyunJin) [reply os/board/common/README.md]: Thanks. I modified it.

File: os/board/common/README.md (hunk 10 above)
Comment (YiHyunJin) [reply os/board/common/README.md]: Thanks. I modified it.

File: os/board/common/README.md (hunk 11 above)
Comment (YiHyunJin) [reply os/board/common/README.md]: thanks.

File: os/board/common/README.md (hunk 12 above)
Comment (YiHyunJin) [reply os/board/common/README.md]: Okay. I modified it.

**Comment by seinfra** [comment]
Target : [0e1bb2870933d7c68ee32c2a7ca6159bc3716cec] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0e1bb2870933d7c68ee32c2a7ca6159bc3716cec] - Code Rule Check (C++) OK.

File: os/board/common/README.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+    2. Append "romfs" at next of *smartfs* to FLASH_PART_TYPE
+
+        ```bash
+        Hardware Configuration -> Board Selection -> append string at Flash partition type list
+        ```
+
+    3. Append "rom" at next of *user* to FLASH_PART_NAME
+
+        ```bash
+        Hardware Configuration -> Board Selection -> append string at FLash partition name list
```
Comment (sunghan-chang) [inline os/board/common/README.md]: FLash -> Flash

File: os/board/common/README.md [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+        ```
+
+    4. Enable the automount config for romfs
+
+        ```bash
+        Hardware Configuration -> Board Selection -> Automount partitions -> [*] Automount romfs partiton
```
Comment (sunghan-chang) [inline os/board/common/README.md]: partiton -> partition

File: os/board/common/README.md [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+
+## How to configure CONFIG
+
+Flash partition is provided by configurations, FLASH_PART_SIZE, FLASH_PART_TYPE and FLASH_PART_NAME. 
+
+Flash partition can be configured the following three parameters:  
+
+ 1. FLASH_PART_SIZE -- represents a partition size  
+ 2. FLASH_PART_TYPE -- represents interface between flash and upper layer  
+ 3. FLASH_PART_NAME (please check above) -- represents partition name
```
Comment (sunghan-chang) [inline os/board/common/README.md]: need to check `(please check above)`

**Comment by seinfra** [comment]
Target : [d8b594dc944b740ee536319c24376b6575df396d] -

**Comment by seinfra** [comment]
Target : [d966b2b3de79970413da27de5704b26a87d4402c] -

**Comment by seinfra** [comment]
Target : [d966b2b3de79970413da27de5704b26a87d4402c] -

**Comment by seinfra** [comment]
Target : [da8397a4a5ba7a5763e319470d2118dd991081fa] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [da8397a4a5ba7a5763e319470d2118dd991081fa] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [bdb0e0bd3f7d0e438a38d9a305b76632c0e1fd96] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [bdb0e0bd3f7d0e438a38d9a305b76632c0e1fd96] - Code Rule Check (C++) OK.

**Pull request merged**